pkgs/development/compilers/llvm/11/compiler-rt/default.nix

```diff
@@ -15,7 +15,8 @@ stdenv.mkDerivation {
   inherit version;
   src = fetch "compiler-rt" "0x1j8ngf1zj63wlnns9vlibafq48qcm72p4jpaxkmkb4qw0grwfy";
 
-  nativeBuildInputs = [ cmake python3 libllvm.dev ];
+  nativeBuildInputs = [ cmake python3 libllvm.dev ]
+     ++ lib.optional stdenv.isDarwin xcbuild.xcrun;
```

Comment:
This won't fly. Hydra doesn't do xcbuild and since this is in the stdenv, we wouldn't get any darwin builds whatsoever.

Reply:
Oh, what prevents hydra from building xcbuild? I tried to make it stdenv-friendly in #185969 by exposing `xcrun` separately, which is just some text files generated by Nix. If that doesn't work, then I guess we'll have to patch the CMake files.

Reply:
The fact that builders don't have no Xcode installed (to my knowledge) and that it's highly impure to run external build tools.

Reply:
Ah, but we don't really use Xcode or the SDKs, just emulate them. For example, our `xcrun` is this shell script: https://github.com/NixOS/nixpkgs/blob/ab265225e8c1abb533f9d650c3f68a090001ebd0/pkgs/development/tools/xcbuild/wrapper.nix#L54-L97

And it references fake macOS SDK and platform directories that are generated plist files. Its entire closure is:

```bash
$ nix-store -qR /nix/store/7142q0hx4a6pxhy01y3iwhdwngv26a36-xcrun
/nix/store/i7ql4rg0667cc3ppp1s0v133cr2414ww-SDKs
/nix/store/kvji81lb8cgbzcj59qv8v5lqwmpz1kss-Platforms
/nix/store/rdsvxa6vz0hj00jl7f7dn2ik14yfl6p9-bash-5.1-p16
/nix/store/7142q0hx4a6pxhy01y3iwhdwngv26a36-xcrun
```

I believe it should be fine, but will try to verify. I thought I was already on a clean macOS, working on a machine provisioned by Scaleway, but it seems Xcode is already on it. I'll delete and try a rebuild.

Reply:
I removed Xcode and Command-Line Tools, cleaned up my Nix store, then did a rebuild, and stdenv builds correctly. (I did find an impurity in my Swift PR, so good thing we tried!)

Reply:
Okay then, I was concerned because it's located under the xcbuild umbrella. Since it isn't really related to xcode, wouldn't it be best to give it a more fitting name and put it somewhere else?

Reply:
I guess it's related in the sense that xcbuild, xcrun, xcode-select, etc all emulate Xcode tooling. It's probably called xcbuild because the core tool its packaging is [facebook/xcbuild](https://github.com/facebookarchive/xcbuild)? And the rest of it is small Nix shims.

But facebook/xcbuild is also unmaintained. It looks like we have several dozen packages using it, though not sure how many use the xcbuild parts, and how many just need the Nix shims. Also not sure how well it is capable of building newer Xcode projects (including Swift stuff).

It might make sense to move the non-xcbuild stuff to a separate top-level attribute, like `xcode-shims`? Maybe @matthewbauer has a stronger opinion. 🙂